src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -5141,6 +5184,234 @@ static synchronized long getColumnEncryptionKeyCacheTtl() {
         return columnEncryptionKeyCacheTtl;
     }
 
+    
+    /**
+     * Used to keep track of an individual handle ready for un-prepare.
+     */
+    private final class PreparedStatementDiscardItem  {
+
+        public int handle; 
+        public boolean directSql;
+
+        public PreparedStatementDiscardItem(int handle, boolean directSql) {
```

Comment:
Question: Can multiple prepare statements share the same handle if they are using the same query? 
Or a new handle is generated for new Prepared Statement? I am trying to understand if a Connection can have multiple prepare statements executing with the same handle. If it can, then do we need have the handles be partitioned / associated with the prepared statement?